src/core/selection/selection.view.js

```diff
@@ -2,43 +2,243 @@ import View from '../view/view';
+			}
+		});
+		model.selectionChanged.watch(e => {
+			if (e.tag.source !== 'toggle'
+				&& e.changes.hasOwnProperty('items')) {
+				this.behavior.select(e.state.items, true);
+			}
 
-				if (isUndefined(state)) {
-					state = this.behavior.state(item);
-					this.behavior.select(item, state === null || !state);
+			if (e.changes.hasOwnProperty('unit')) {
+				if (e.changes.unit.oldValue === 'row') {
+					e.state.items.forEach((item) => this.blurRow(item));
+				} else if (e.changes.unit.oldValue === 'column') {
+					//blur all columns
 				}
-				else {
-					this.behavior.select(item, state);
+			}
+			if (e.changes.hasOwnProperty('unit') || e.changes.hasOwnProperty('mode')) {
+				this.behavior = behaviorFactory(model);
+				model.navigation({
+					column: -1,
+					row: -1
+				});
+				if (e.state.unit === 'row') {
+					e.state.items.forEach((item) => {
+						this.blurRow(item);
+					});
+				} else if (e.state.unit === 'column') {
+					//blur all highlighted columns
+				}
+			}
+			if (e.changes.hasOwnProperty('items')) {
+				if (e.state.unit === 'row') {
+					e.changes.items.newValue.forEach((item) => {
+						if (this.state(item)) {
+							this.highlightRow(item);
+						}
+					});
+					e.changes.items.oldValue.forEach((item) => {
+						if (!this.state(item)) {
+							this.blurRow(item);
+						}
+					});
+				} else if (e.state.unit === 'column') {
+					//blur oldValue columns
+					//highlight newValue columns
 				}
-
-				model.selection({items: this.behavior.view}, {source: 'toggle'});
 			}
 		});
+	}
 
-		model.viewChanged.watch(() => this.behavior = behaviorFactory(model));
+	highlightRow(item) {
+		const index = this.model.view().rows.indexOf(item);
+		const body = this.markup.body;
+		if (body && body.rows[index]) {
+			for (let cell of body.rows[index].cells) {
+				this.highlight(cell);
+			}
+		}
+	}
 
-		model.selectionChanged.watch(e => {
-			if (e.tag.source !== 'toggle' && e.changes.hasOwnProperty('items')) {
-				this.behavior.select(e.state.items, true);
+	blurRow(item) {
+		const index = this.model.view().rows.indexOf(item);
+		const body = this.markup.body;
+		if (body && body.rows[index]) {
+			for (let cell of body.rows[index].cells) {
+				this.blur(cell);
 			}
+		}
+	}
 
-			if (e.changes.hasOwnProperty('unit') ||
-				e.changes.hasOwnProperty('mode')) {
-				this.behavior = behaviorFactory(model);
+	highlightColumn(index) {
+		const body = this.markup.body;
+		if (body && body.rows) {
+			for (let row of body.rows) {
+				this.highlight(row.cells[index]);
 			}
-		});
+		}
+	}
+
+	blurColumn(index) {
+		const body = this.markup.body;
+		if (body && body.rows) {
+			for (let row of body.rows) {
+				this.blur(row.cells[index]);
+			}
+		}
+	}
+
+	highlight(item) {
+		item.classList.add(`${GRID_PREFIX}-selected`);
+	}
+
+	blur(item) {
+		item.classList.remove(`${GRID_PREFIX}-selected`);
+	}
+
+	get commands() {
+		const model = this.model;
```

Comment:
Consider to move commands to separate file. Kind of `commandsFactory = model => ...`